torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1818,74 +1818,97 @@ def sample_inputs_cat_concat(op_info, device, dtype, requires_grad, **kwargs):
-                     error_regex=err_msg)
+                     error_regex='Tensors must have same number of dimensions: got 4 and 2')
 
     # error inputs for different devices
     if torch.device(device).type == 'cuda':
         x_cuda = make_tensor((3, 3), device=device, dtype=torch.float32)
         y_cpu = make_tensor((3, 3), device='cpu', dtype=torch.float32)
-        yield ErrorInput(SampleInput((x_cuda, y_cpu)),
-                         error_regex='Expected all tensors to be on the same device')
+        sample_input = SampleInput((x_cuda, y_cpu))
+        if is_ref:
+            yield ErrorInput(sample_input,
+                             error_regex=('Expected all tensors to be on the same device, '
+                                          'but found at least two devices, cuda:0 and cpu!'))
+        else:
+            yield ErrorInput(sample_input,
+                             error_regex=(r'Expected all tensors to be on the same device, '
+                                          r'but found at least two devices, cuda:0 and cpu! '
+                                          r'\(when checking argument for argument tensors in '
+                                          r'method wrapper_cat\)'))
 
     # error inputs for different input sizes for more than 2 tensors
     yield ErrorInput(SampleInput([make_arg((L, 1)), make_arg((L, 1, 1)), make_arg((L, 1, 1))]),
-                     error_regex='Tensors must have same number of dimensions')
+                     error_regex='Tensors must have same number of dimensions: got 2 and 3')
 
     yield ErrorInput(SampleInput([make_arg((S, 1, M)), make_arg((S, 1, 1)), make_arg((S, M, 1))],
                                  kwargs={'dim': 1}),
-                     error_regex='Sizes of tensors must match')
+                     error_regex=('Sizes of tensors must match except in dimension 1. '
+                                  'Expected size 10 but got size 1 for tensor number 1 in the list'))
 
     # error inputs for None input
+    if is_ref:
+        none_regex = "expected Tensor as element 1 in argument 0, but got <class 'NoneType'>"
+    else:
+        none_regex = 'expected Tensor as element 1 in argument 0, but got NoneType'
     yield ErrorInput(SampleInput((make_arg((S, 1, 1)), None)), error_type=TypeError,
-                     error_regex='got None')
+                     error_regex=none_regex)
 
     # error inputs for zero-dimensional tensors
     yield ErrorInput(SampleInput([make_arg(()), make_arg(())]),
-                     error_regex='zero-dimensional.*cannot be concatenated')
-
-    # error inputs for different dtype of out tensors
-    d = make_tensor((2, 3), device=device, dtype=torch.double)
-    x = make_tensor((2, 3), device=device, dtype=torch.float32)
-    yield ErrorInput(SampleInput(x, kwargs={'out': d}), error_type=TypeError,
-                     error_regex='invalid combination of arguments')
```

Comment:
This was failing not because of different dtypes, which is allowed. It was failing because `x` is not a sequence, which is not supported by available overloads. So I removed this error input.